### PR TITLE
rfc: proposed cleanup

### PR DESCRIPTION
**Pull request by jdknight**
I'd like to propose a series of cleanup changes to the `sphinxcontrib-confluencebuilder` repository. The key changes include:

* Namespace cleanup for builder/writer.
* Add missing unit test for "use macros for admonitions" #11.
* Cleanup committed version references and improve bumpversion configuration.
* Populate changelog information.
* Remove/cleanup `test_builder` output files.

### CODE REVIEW

**Comment by tonybaloney** [comment]
thanks @jdknight 

this project was started from the sphinx plugin template, I hadn't ventured too far outside the namespaces that were provided. 
Have you verified that the Sphinx plugin loader supports submodules?

**Comment by tonybaloney** [comment]
fyi, I've added you as a project collaborator. I can also add you on PyPi if you have an account.

**Comment by jdknight** [comment]
@tonybaloney 

> Have you verified that the Sphinx plugin loader supports submodules?

I believe so (if you're talking about just running this source with `sphinx`; successful build with `test_builder`), but I was looking for an second opinion from you just in case. Been awhile since I'd dabbled in Python so re-learning about new tools, processes, etc. (ex. I've never use `tox` before). I've seen the pattern followed by a series of sphinx-contrib repositories ([doxylink](https://bitbucket.org/birkenfeld/sphinx-contrib/src/e758073384ef/doxylink/), [feed](https://bitbucket.org/birkenfeld/sphinx-contrib/src/e758073384efd1ed5ed1e6286301b7bef71b27cf/feed/), [googlechart](https://bitbucket.org/birkenfeld/sphinx-contrib/src/e758073384efd1ed5ed1e6286301b7bef71b27cf/googlechart/), etc.), so I figured it would be fine.

My intention for doing this all stems from trying to gracefully place all the source files into a single location. I was planning to submit `ConfluencePublisher` class instance in a [future commit](https://github.com/jdknight/sphinxcontrib-confluencebuilder/commit/7885c5f67abdd0e05241bfa63057c3b3de0008ae). I figured it would be nice to have it contained under `sphinxcontrib/confluencebuilder` instead of something like `sphinxcontrib/publishers`; however, you'll know best.

> fyi, I've added you as a project collaborator.

Neat! 😍

Never been in a collaborator position on a GitHub project I didn't own. Any policy, contributors notes, etc. I should consider? For the interim I'll just be submitting pull requests for now to get my feet wet with proper Python management.

> I can also add you on PyPi if you have an account.

Cool; just [made an account](https://pypi.python.org/id/jdknight) (something else I'll refrain from touching till I learn all the ins and outs).

**Comment by tonybaloney** [comment]
I've added you as a maintainer on PyPi too, so you can release new versions.

for now, my release process is- 

```bash
bumpversion minor
rm -f dist/*
python setup.py sdist bdist_wheel
twine upload dist/*
```

I saw you updated some of the bumpversion configuration. 

I maintain *too* many packages on PyPi and since you're now the biggest contributor to the project I'd be happy for you to co-maintain the package.
Check out any of my other packages to copy patterns on testing, e.g.
https://github.com/tonybaloney/requests-staticmock

File: setup.cfg [hunk 1]
```diff
@@ -1,12 +1,18 @@
 [bumpversion]
-current_version = 0.5.0
+current_version = 0.6.0rc
```
Comment (tonybaloney) [inline setup.cfg]: this wouldn't meet the PEP for versioning. it would be 0.6.0rc0

**Comment by tonybaloney** [comment]
also, `.gitignore` should contain the output directories that sphinx builds for the test suite. This would stop them appearing in `git status` every time.

**Comment by jdknight** [comment]
@tonybaloney 

> this wouldn't meet the PEP for versioning

I've updated this pull request to use proper PEP versioning (see: 87f7218; 477f758). Are you fine with having the "development release" post-fix or would you prefer to keep a simpler versioning schema?

>  `.gitignore` should contain the output directories that sphinx builds for the test suite

Just for clarification, are you indicating that:

- The test output should already be covered by the existing `.gitignore`; or are you suggesting that,
- I should not have introduced the `test/testproj/build` directory and instead added an explicit ignore option for `test/testproj/doctree` and `test/testproj/out`?

When I manually invoked the test, I still saw the generated output in my Git client (before this change). If your point is the latter, do you want me to revert the directory change and instead at the explicit ignore option (or is the existing change fine)?

> I'd be happy for you to co-maintain the package

Glad to help and glad to be allowed to help. Maybe I can help try to address [some issues](https://github.com/tonybaloney/sphinxcontrib-confluencebuilder/milestone/1), generate a v0.6 release and you can comment along the way to catch anything I do wrong?  :blush:

**Comment by tonybaloney** [comment]
This PR is good to merge. 👍

**Pull request merged**